sdk/core/System.ServiceModel.Rest/src/PipelineMessage.cs

```diff
@@ -5,21 +5,45 @@
 
 namespace System.ServiceModel.Rest.Core;
 
-public abstract class PipelineMessage : IDisposable
+public class PipelineMessage : IDisposable
 {
-    protected PipelineMessage(PipelineRequest request, ResponseErrorClassifier classifier)
+    private readonly PipelineRequest _request;
+    private readonly ResponseErrorClassifier _classifier;
+
+    private PipelineResponse? _response;
+
+    protected internal PipelineMessage(PipelineRequest request, ResponseErrorClassifier classifier)
     {
-        PipelineRequest = request;
-        ResponseErrorClassifier = classifier;
+        _request = request;
+        _classifier = classifier;
     }
 
     public CancellationToken CancellationToken { get; set; } = CancellationToken.None;
 
-    public abstract PipelineResponse? PipelineResponse { get; set; }
+    public virtual PipelineRequest Request { get => _request; }
 
-    public abstract PipelineRequest PipelineRequest { get; set; }
+    public virtual PipelineResponse Response
+    {
+        get
+        {
+            if (_response is null)
+            {
+                throw new InvalidOperationException("Response was not set.");
+            }
 
-    public abstract ResponseErrorClassifier ResponseErrorClassifier { get; set; }
+            return _response;
+        }
+
+        set => _response = value;
```

Comment:
TODO: does this need to be settable?

Reply:
I looked at this and it's a lot easier to preserve the APIs that are inherited, and it may be needed for mocking.